router/src/main.rs

```diff
@@ -374,8 +382,49 @@ async fn main() -> Result<(), RouterError> {
         }
     };
 
+    // Only send usage stats when TGI is run in docker
+    let is_docker = option_env!("DOCKER_LABEL").is_some();
```

Comment:
I think this needs env variable to be set in container. For a more reliable way of detecting container, you probably need to look at init process groups to check if `cpuacct` is mounted on `/` or not.

Reply:
Good point! 👍 

After a small read online I found that this was quite a common way. Do you know if there are any downsides/benefits in checking the `cgroup` vs `cpuacct`?

```rust
pub fn is_container() -> io::Result<bool> {
    let path = Path::new("/proc/self/cgroup");
    let file = File::open(&path)?;
    let reader = io::BufReader::new(file);

    for line in reader.lines() {
        let line = line?;
        // Check for common container runtimes
        if line.contains("/docker/") || line.contains("/docker-") ||
           line.contains("/kubepods/") || line.contains("/kubepods-") ||
           line.contains("containerd") || line.contains("crio") ||
           line.contains("podman") {
            return Ok(true);
        }
    }
    Ok(false)
}
```

Reply:
`cpuacct` was one option to check in `cgroup` but I like your solution better as you check for them all 👍

Reply:
🫡